app/Http/Controllers/Backend/PermissionsController.php

```diff
@@ -0,0 +1,297 @@
+<?php
+namespace App\Http\Controllers\Backend;
+
+use Illuminate\Http\Request;
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+use App\Models\Backend\Permission;
+
+class PermissionsController extends Controller
+{
+    /**
+     * Display a listing of the Permissions.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $data['data'] = Permission::all();
+        return view('backend.permissions.index')->with($data);
+    }
+
+    /**
+     * Show the form for creating a new Permissions.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $modules['modules'] = $this->getModule();
+        return view('backend.permissions.create')->with($modules);
+    }
+
+    /**
+     * Store a newly created Permissions in storage.
+     *
+     * @param \Illuminate\Http\Request $request request create
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        // check permission is exists?
+        if (!$this->checkPermissionExisted($request)) {
+            // create new permission
+            $newRole = new Permission;
+            $newRole->module = $request->module;
+            
+            // set permission
+            $arrRole = $this->setPermission($request);
+            $newRole->see     = $arrRole['see'];
+            $newRole->inset   = $arrRole['inset'];
+            $newRole->edit    = $arrRole['edit'];
+            $newRole->destroy = $arrRole['destroy'];
+
+            // check save successfull?
+            if ($newRole->save()) {
+                $request->session()->flash('message', trans('labels.MessageAddRoleOk'));
```

Comment:
The key in translation file should be managed carefully. `'labels.MessageAddRoleOk'` is not a good key. My idea is separate to 2 files. `labels` and `messages`. 
- `labels` : only for `html` content
- `messages` : for message from controller.
then 
`trans('messages.role_added_successfully');